sdk/compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Error.hs

```diff
@@ -266,9 +270,88 @@ instance Pretty WarnableError where
       vsep
         [ "This package defines both exceptions and templates. This may make this package and its dependents not upgradeable."
         , "It is recommended that exceptions are defined in their own package separate from their implementations."
-        , "Ignore this error message with the --warn-bad-exceptions=yes flag."
+        , "Ignore this error message with the -Wupgrade-exceptions flag."
         ]
 
+data DamlWarningFlagStatus
+  = AsError -- -Werror=<name>
+  | AsWarning -- -W<name>
+  | Hidden -- -Wno-<name>
+
+data DamlWarningFlag
+  = RawDamlWarningFlag
+    { rfName :: String
+    , rfStatus :: DamlWarningFlagStatus
+    , rfFilter :: WarnableError -> Bool
+    }
+  | WarnBadInterfaceInstances Bool -- When true, same as -Wupgrade-interfaces
+  | WarnBadExceptions Bool
```

Comment:
```suggestion
  | WarnBadExceptions Bool -- When true, same as -Wupgrade-exceptions
```
To match?

Reply:
I've generally improved the comments on these